mantidimaging/gui/windows/main/view.py

```diff
@@ -462,18 +483,18 @@ def show_live_viewer(self, live_data_path: Path) -> None:
             self.live_viewer.show()
 
     @property
-    def stack_list(self):
+    def stack_list(self) -> list[StackVisualiserView]:
         return self.presenter.stack_visualiser_list
 
     @property
-    def strict_dataset_list(self):
```

Comment:
@samtygier-stfc In the MainWindow class, properties such as stack_list and strict_dataset_list are expected to return lists of StackVisualiserView and StrictDataset respectively. However, the corresponding properties in the MainWindowPresenter class return lists of StackId and DatasetId. This mismatch leads to type errors because the expected return types in MainWindow do not match the actual return types from the presenter.

Reply:
Looks like it should be `list[DatasetId]` here, and that needs to be propagated into a few other places